### DISCUSSION TITLE
When can we pass `SubscriptOps` to setter?

### DISCUSSION THREAD

**Discussion by nnethery** [jjjkkkjjj/Matft#37] [Ideas]
When do you think you will provide support to pass `SubscriptOps` in setters? I would like to do something like `loss_matrix[:, :, 0] = f_losses[:, :, 0]`. How much effort will this take? I could do it if it looks like it won't be too much work, however I would have to learn the Accelerate framework.

**Comment by jjjkkkjjj** [comment]
@nnethery 
Sorry for late response!
I think I've already developed the operation you mentioned.
I created the [new tests](https://github.com/jjjkkkjjj/Matft/blob/6a47a8656cd6e55d259c05bda3f367595b2d80f4/Tests/MatftTests/SubscriptTest.swift#L939), please check them again.

Example;

```swift
let a = Matft.arange(start: 0, to: 27, by: 1, shape: [3,3,3])
let b = Matft.arange(start: -27, to: 0, by: 1, shape: [3,3,3])
a[Matft.all, Matft.all, 0] = b[Matft.all, Matft.all, 0]
print(a)
/*
mfarray = 
[[[	-27,		1,		2],
[	-24,		4,		5],
[	-21,		7,		8]],

[[	-18,		10,		11],
[	-15,		13,		14],
[	-12,		16,		17]],

[[	-9,		19,		20],
[	-6,		22,		23],
[	-3,		25,		26]]], type=Int, shape=[3, 3, 3]
*/
```

Here is `numpy` result;

```python
>>> a = np.arange(27).reshape(3,3,3)
>>> b = np.arange(-27,0).reshape(3,3,3)
>>> a[:,:,0] = b[:,:,0]
>>> a
array([[[-27,   1,   2],
        [-24,   4,   5],
        [-21,   7,   8]],

       [[-18,  10,  11],
        [-15,  13,  14],
        [-12,  16,  17]],

       [[ -9,  19,  20],
        [ -6,  22,  23],
        [ -3,  25,  26]]])

```

**Comment by nnethery** [comment]
Thank you! I think I figured out what my problem was. I changed my project in Xcode from using Matft's version tag `0.3.2` to `master` branch and it ended up working!
<img width="567" alt="Screenshot 2023-01-08 at 3 32 22 PM" src="https://user-images.githubusercontent.com/11582381/211217744-e1121500-7842-4e3d-a792-9c5256c6030d.png">
